src/sentry/integrations/slack/tasks.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import absolute_import
+
+import six
+
+from django.conf import settings
+
+from sentry import http
+from sentry.utils import json
+from sentry.tasks.base import instrumented_task
+from sentry.models import Integration, Project, Rule
+
+from sentry.utils.redis import redis_clusters
+
+MEMBER_PREFIX = "@"
+CHANNEL_PREFIX = "#"
+
+LIST_TYPES = [
+    ("channels", "channels", CHANNEL_PREFIX),
+    ("groups", "groups", CHANNEL_PREFIX),
+    ("users", "members", MEMBER_PREFIX),
+]
+
+
+class RedisRuleStatus(object):
+    def __init__(self, uuid):
+        self.uuid = uuid
+
+        cluster_id = getattr(settings, "SENTRY_RULE_TASK_REDIS_CLUSTER", "default")
+        self.client = redis_clusters.get(cluster_id)
+        self._set_value("pending")
+
+    def _get_redis_key(self):
+        return u"slack-channel-task:1:{}".format(self.uuid)
+
+    def _set_value(self, status, rule_id=None):
+        value = self._format_value(status, rule_id)
+        self.client.set(self._get_redis_key(), u"{}".format(value), ex=60 * 60)
+
+    def _format_value(self, status, rule_id):
+        value = {"status": status}
+        if rule_id:
+            value["rule_id"] = six.text_type(rule_id)
+        if status == "failed":
+            # TODO(meredith): set some generic error? or pass in an error?
+            value["error"] = "some error"
+
+        return json.dumps(value)
+
+
+@instrumented_task(name="sentry.integrations.slack.search_channel_id", queue="integrations")
+def find_channel_id_for_rule(serializer, project_id, uuid, rule_id=None):
```

Comment:
Passing the serializer to the task like this is pretty risky. We have to pickle the object, and so now, if someone changes the code for the serializer or moves it then this task will break. I don't think we should do this.  I think you'd be better off passing the state that you initially passed to the serializer here and then reconstructing it. Or maybe just passing the `validated_data` through since that's all you really need from the serializer.

Even better might be to abstract the rule creation logic out of the rule serializer and use that here instead.

Reply:
> We have to pickle the object, and so now, if someone changes the code for the serializer or moves it then this task will break.

Didn't think about this. Not sure the best way to abstract out the rule creation logic yet, can maybe start with passing through the `validated_data`

Reply:
Yeah, we could also just pass all the validated values as parameters, rather than a json blob. Might be easier to understand what is required, but I'll leave that to you to decide.